AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -249,7 +288,13 @@ - (void)startUpdate {
       return;
     }
     NSString *distributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
-    if (updateToken || distributionGroupId) {
+    @synchronized(self) {
+      if (self.updateFlowInProgress) {
+        MSLogDebug([MSDistribute logTag], @"Previous update flow is in progress. Ignore the request.");
```

Comment:
There is something i don't understand about this message. It says it'll "Ignore the request" but the flow continuous.

Reply:
Yeah, shouldn't this return?  Or if not, then `self.updateFlowInProgress` should be used in control flow later, I would think

Reply:
`return` was missing.